HISTORY.md

```diff
@@ -3,6 +3,9 @@
 ### New Features
 * Allow WriteBatchWithIndex to index a WriteBatch that includes keys with user-defined timestamps. The index itself does not have timestamp.
 
+### Bug Fixes
+* Fixed a data race on `VersionSet` in `DBImpl::ResumeImpl()`
```

Comment:
Nit: illegal (stale) memory access on 'versions_' caused by race between db close issued by ... and `DBImpl::ResumeImpl()`.
The '...' can be something like "user-defined listener's OnErrorRecoveryXXX() methods", but not sure what is a good succint way of describing it.

Reply:
Good point! And thank you for providing an example. 

I also found it hard to write HISTORY.md for fix like this since (a) it's not as visible as fixing a bug directly impacting user API (so I am not even sure whether to include this in HISTORY.md) and (b)it involves internal implementation logic (so I am not sure how much detail I should diclose to users).

Reply:
Yeah, we can keep the entry in the HISTORY.md simple, just reference this PR so that people can conveniently know where to look for more context. For example
```
* Fixed a data race on `versions_` between `DBImpl::ResumeImpl()` and `EventListener::OnErrorRecoveryXX()` (#9496)
```

Reply:
Fixed